### PR TITLE
Add dark mode

### PR DESCRIPTION
**Pull request by lukebemish**
This adds a dark mode using the relevant `@media` css by default, with a button to toggle on desktop
image

------------------
Preview URL: https://pr-8.neoforged-website-previews.pages.dev

### CODE REVIEW

**Comment by lukebemish** [comment]
Note that this switches some of the style stuff around to separate out the color and other parts of it, and to use partials to share most of the styling between the two themes.

**Comment by ChampionAsh5357** [comment]
How can you switch between the two modes?

**Comment by lukebemish** [comment]
> How can you switch between the two modes?

It uses the CSS `@media` query - so whatever your browser asks for, which is a pretty common approach. Something to manually switch could be added later in theory, but that gets more complicated

**Comment by ChampionAsh5357** [comment]
Okay, I was thinking since we have something for the docs that can switch manually, there should be one for the webpage. But, I will acquiesce for now.

**Comment by ChampionAsh5357** [comment]
image

After a quick look through the pages, my only comment is that the buttons for github and mastodon embeds doesn't look that great. Also, there's a line underneath that's not seen in light mode.

**Comment by cpw** [comment]
HAWT

**Comment by lukebemish** [comment]
> Okay, I was thinking since we have something for the docs that can switch manually, there should be one for the webpage. But, I will acquiesce for now.

Not a bad idea - I've got the infrastructure set up so that it's all in separate stylesheets, so that could probably be added in the future with relative ease

**Comment by lukebemish** [comment]
Added a mode toggle button. It should all default to the system preference, unless you hit the button

**Comment by ChampionAsh5357** [comment]
I'll do a proper code review when I get home. Until then, can you fix the mixing tabs and spaces in the files you created?

Looks great!

**Comment by lukebemish** [comment]
Fixed; that was actually because it was mixed tabs and spaces, as VSCode doesn't handle that well without an editorconfig or something, and GH renders tabs as 8 spaces

File: hugo.toml [hunk 1]
```diff
@@ -28,9 +28,4 @@ theme = "mainroad"
 [Params.widgets]
   recent_num = 5 # Set the number of articles in the "Recent articles" widget
   tags_counter = false # Enable counter for each tag in "Tags" widget (disabled by default)
-
-[Params.style]
-  vars.highlightColor = "#d7742f"
-  vars.backgroundColor = "#d5e0df"
-  vars.boxColor = "#bdc7c7"
-  vars.borderColor = "#aab0b3"
+
```
Comment (ChampionAsh5357) [inline hugo.toml:31]: Add EOL

**Review by ChampionAsh5357 — COMMENTED** [review]
Looks fine at a glance. I don't understand the separation between what goes in the colors css compared to the style.css since it seems to be a mix of both conditional and non-conditional data.

**Comment by lukebemish** [comment]
The difference is literally just "handles stuff that might have to do with colors or needs to go after colors" and "doesn't". Made it easier to work on when I was changing around what was conditional and what wasn't

**Comment by lukebemish** [comment]
Let me know if there's any changes you'd like from me on this; it should be ready to go as far as I'm aware

**Comment by ChampionAsh5357** [comment]
Yep yep. I think it's fine. I'm letting anyone else on the team do any checks of their own. If no one comments, it'll be merged in about 6ish hours.

**Comment by cpw** [comment]
This looks great to me @ChampionAsh5357 i'm ok with merging :+1:

**Pull request merged**